src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -118,6 +119,22 @@ private BoundExpression ApplyConvertedTypes(BoundExpression expr, TupleBinaryOpe
                 analysisResult.LeftConversion, analysisResult.RightConversion, signature.Method, conversionIntoBoolOperator, boolOperator);
         }
 
+        private bool TypelessNewHasAnyErrors(BoundExpression left, BoundExpression right)
```

Comment:
This is probably not the best thing to do.. thoughts?

Reply:
Just to clarify, this is to handle a scenario like `(1, 2) == (1, new())`, right?
If so, I don't understand why we need this special conversion check. In a tuple comparison, the conversion should already be added (same conversions as needed to do `2 == new()`).

---
In reply to: [209431545](https://github.com/dotnet/roslyn/pull/29167#discussion_r209431545) [](ancestors = 209431545)

Reply:
The problem is that `new()` converts to any type (so foundOperator is always true) as a result we don't propagate errors of an invalid `new()` usage on the tuple equality (e.g. if no applicable ctor was found) . The actual diagnostics are discarded when we're binding the operand conversions.

http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_TupleOperators.cs,30

Reply:
It sounds like the relevant scenario is `(1, 2) == new()` then.
If so, a possible fix is to not discard the conversion diagnostics (line 30, which you pointed at), with something like: `ApplyConvertedTypes(right, operators, isRight: true, rightIsNew ? diagnostics : discardDiagnostics);`

Reply:
that would be valid now - it calls tuple's default ctor. the affected scenario is `("", 1) == (new(), 1)`. perhaps we should iterate over all diagnostics and extract out relevant ones? still doesn't seem to be right. somehow we should keep those errors on the resulting node after conversion.